components/LocationInput.tsx

```diff
@@ -82,6 +82,27 @@ const MapBoxInput: FC<PropsWithChildren<MapBoxInputProps>> = ({
         setIsSubmissionEnabled(map.current!.getZoom() > zoomLevelThreshold);
         updateMarker(marker, map.current!, zoomLevelThreshold, newLng, newLat);
       });
+
+      return () => {
+        map?.current?.off('move', () => {
+          const newLng = map.current!.getCenter().lng;
+          const newLat = map.current!.getCenter().lat;
+
+          setLng(newLng);
+          setLat(newLat);
+          setLngExternal(newLng);
+          setLatExternal(newLat);
+          setZoom(map.current!.getZoom());
+          setIsSubmissionEnabled(map.current!.getZoom() > zoomLevelThreshold);
+          updateMarker(
+            marker,
+            map.current!,
+            zoomLevelThreshold,
+            newLng,
+            newLat
+          );
+        });
```

Comment:
If you move the arrow function passed into the `off` call (the lines this comment refers to) to a separate declaration in the useEffect and name it `handler` or something, you can then use the same declaration when `on` is called a few lines above. This will effectively remove the handler when the component unmounts, and might fix the memory leak issue?

Reply:
oops i hadn't pushed yet this is very old

Reply:
also doing that actually funnily just breaks the map—the marker just fails to show up

Reply:
Dang alright